test/runtests.jl

```diff
@@ -54,11 +85,57 @@ end
     atmosphere = SSSynth.read_model_atmosphere("data/sun.krz")
     @test length(atmosphere) == 56
     @test issorted(first.(atmosphere))
-    @test atmosphere[1].tau == 9.747804143e-3
+    @test atmosphere[1].colmass == 9.747804143e-3
     @test atmosphere[1].temp == 4066.8
     @test atmosphere[1].electron_density == 3.76980e10
     @test atmosphere[1].number_density == 4.75478e14
     @test atmosphere[1].density == 1.00062e-9
 end
 
-include("continuum_opacity.jl")
+@testset "synthesis" begin
+
+    @testset "calculate absolute abundances" begin
+        @test_throws ArgumentError SSSynth.get_absolute_abundances(["H"], 0.0, Dict("H"=>13))
+
+        @testset for metallicity in [0.0, 1.0], A_X in [Dict(), Dict("C"=>9)]
+            for elements in [SSSynth.atomic_symbols, ["H", "He", "C", "Ba"]]
+                nxnt = SSSynth.get_absolute_abundances(elements, metallicity, A_X)
+
+                #abundances for the right set of elementns
+                @test Set(elements) == Set(keys(nxnt))
+
+                #correct absolute abundances?
+                if "C" in keys(A_X)
+                    @test log10(nxnt["C"]/nxnt["H"]) + 12 ≈ 9
+                end
+                @test log10(nxnt["He"]/nxnt["H"]) + 12 ≈ SSSynth.solar_abundances["He"]
+                @test log10(nxnt["Ba"]/nxnt["H"]) + 12 ≈ 
+                    SSSynth.solar_abundances["Ba"] + metallicity
+
+                #normalized?
+                if elements == SSSynth.atomic_symbols
+                    @test sum(values(nxnt)) ≈ 1
+                else
+                    @test sum(values(nxnt)) < 1
+                end
+            end
+        end
+    end
+
+    @testset "number densities" begin
+        abundances = Dict("C"=>1e-8, "H"=>0.9)
+        nₜ = 1e15
+        n = SSSynth.per_species_number_density(nₜ, nₜ*1e-3, 4500.0, abundances)
+        @test Set(keys(n)) == Set(["C_I", "C_II", "C_III", "H_I", "H_II"])
+        @test n["C_III"] < n["C_II"] < n["C_I"] < n["H_II"] < n["H_I"]
+        @test n["C_III"] + n["C_II"] + n["C_I"] ≈ abundances["C"] * nₜ
+        @test n["H_II"] + n["H_I"] ≈ abundances["H"] * nₜ
```

Comment:
Dumb question, what kind of tolerance does the `≈` symbol use? This isn't super important

Reply:
I believe it's machine precision by default but I'm not sure.  It's definitely below the precision we are going for.